### PR TITLE
rgw: use new Stopped state for special handling of 'bucket sync disable'

### PR DESCRIPTION
**Pull request by cbodley**
some sync modules like pubsub can disable full sync, and only run incremental sync. without full sync, 'bucket sync disable/enable' will continuously restart incremental sync from the beginning of the bilog
    
this adds a new Stopped state for 'bucket sync disable', which resets the incremental sync position to the end of the bilog, instead of the beginning. this allows it to continue processing from the end, though it may miss some entries

Fixes: https://tracker.ceph.com/issues/43768

### CODE REVIEW

File: src/rgw/rgw_data_sync.cc [hunk 1]
```diff
@@ -2776,8 +2776,16 @@ class RGWInitBucketShardSyncStatusCoroutine : public RGWCoroutine {
             status.state = rgw_bucket_shard_sync_info::StateFullSync;
             status.inc_marker.position = info.max_marker;
           } else {
+            const bool reinit = status.state == rgw_bucket_shard_sync_info::StateReInit;
+            if (reinit) {
+              // continue processing at the end, though we may skip entries
+              // after the CLS_RGW_OP_RESYNC
+              status.inc_marker.position = info.max_marker;
```
Comment (cbodley) [inline src/rgw/rgw_data_sync.cc]: ideally we wouldn't have to set this at the end, we could just leave it at the SYNCSTOP position. then we'd resume from the RESYNC op and sync everything after. need to test whether that works reliably

**Comment by yuvalif** [comment]
fix was manually verified with the test added in: https://github.com/ceph/ceph/pull/33068

**Comment by cbodley** [comment]
i'm still seeing test failures from this in http://pulpito.ceph.com/cbodley-2020-02-04_17:23:30-rgw:multisite-wip-rgw-pubsub-syncstop-distro-basic-smithi/

it's not handling the case where we reinit and find syncstopped is still true - in that case, we delete the sync status object, which defaults back to incremental with an empty position

**Comment by cbodley** [comment]
oh wow, it's all green: http://pulpito.ceph.com/cbodley-2020-02-06_14:33:45-rgw:multisite-wip-rgw-pubsub-syncstop-distro-basic-smithi/

@yehudasa can you please help with review?

**Pull request merged**